### ISSUE TITLE
should reject type matches with unknown types

### ISSUE THREAD

**Issue by lutzhamel** [asteroid-lang/asteroid#92]
Currently Asteroid rejects the following program
```
let x:%Foo = 1.
```
with the error message,
```
pattern match failed: expected typematch Foo got an object of type integer
```
which is the wrong error message.  It should reject the statement based on the fact that `Foo` has not been defined as a type.

**Comment by olwmc** [comment]
Since this is getting caught in the [final else clause in the typematch branch](https://github.com/lutzhamel/asteroid/blob/d8d412b6acf4bd4ad4f47ff7f66c53969c033779/code/asteroid_walk.py#L222), we could add a catch in there to check if it's this type of error. I.E.

```python
# ttc
# Should we have an else here?
else:
        if not state.symbol_table.find_sym(typematch):
            raise PatternMatchFailed( "{} is not a valid type for typematch".format(typematch) )
        else:
            raise PatternMatchFailed(
                "expected typematch {} got an object of type {}"
                .format(typematch, term[0]))
```

Though this might need to be given more thought, this solution doesn't seem to regress any already succeeding typematches.

**Comment by lutzhamel** [comment]
@olwmc I agree.  But the check has to go deeper than that because this would pass your proposed solution,
```
let Foo = 1.
let x:%Foo = 1.
```
You also have to check that `Foo` is indeed a typename.

**Comment by olwmc** [comment]
Great catch!!

**Comment by olwmc** [comment]
Closed with #108
